test/parallel_api/iterator/tuple_unit.pass.cpp

```diff
@@ -0,0 +1,100 @@
+
+    // Test binary comparison operators for std::tuple and oneAPI::dpl::__internal::tuple
+    EXPECT_EQ((t1 == t2), (onedpl_t1 == onedpl_t2), "equality comparison does not match std::tuple");
+    EXPECT_EQ((t1 != t2), (onedpl_t1 != onedpl_t2), "inquality comparison does not match std::tuple");
+    EXPECT_EQ((t1 < t2), (onedpl_t1 < onedpl_t2), "less than comparison does not match std::tuple");
+    EXPECT_EQ((t1 <= t2), (onedpl_t1 <= onedpl_t2), "less than or equal to comparison does not match std::tuple");
+    EXPECT_EQ((t1 > t2), (onedpl_t1 > onedpl_t2), "greater than comparison does not match std::tuple");
+    EXPECT_EQ((t1 >= t2), (onedpl_t1 >= onedpl_t2), "greater than or equal to comparison does not match std::tuple");
+
+    EXPECT_EQ((t1 == t2), (t1 == onedpl_t2), "equality comparison does not match std::tuple");
+    EXPECT_EQ((t1 != t2), (t1 != onedpl_t2), "inquality comparison does not match std::tuple");
+    EXPECT_EQ((t1 < t2), (t1 < onedpl_t2), "less than comparison does not match std::tuple");
+    EXPECT_EQ((t1 <= t2), (t1 <= onedpl_t2), "less than or equal to comparison does not match std::tuple");
+    EXPECT_EQ((t1 > t2), (t1 > onedpl_t2), "greater than comparison does not match std::tuple");
+    EXPECT_EQ((t1 >= t2), (t1 >= onedpl_t2), "greater than or equal to comparison does not match std::tuple");
+
+    EXPECT_EQ((t1 == t2), (onedpl_t1 == t2), "equality comparison does not match std::tuple");
+    EXPECT_EQ((t1 != t2), (onedpl_t1 != t2), "inquality comparison does not match std::tuple");
+    EXPECT_EQ((t1 < t2), (onedpl_t1 < t2), "less than comparison does not match std::tuple");
+    EXPECT_EQ((t1 <= t2), (onedpl_t1 <= t2), "less than or equal to comparison does not match std::tuple");
+    EXPECT_EQ((t1 > t2), (onedpl_t1 > t2), "greater than comparison does not match std::tuple");
+    EXPECT_EQ((t1 >= t2), (onedpl_t1 >= t2), "greater than or equal to comparison does not match std::tuple");
+
+    auto onedpl_t3 = to_onedpl_tuple(t1);
+    auto onedpl_t4 = to_onedpl_tuple(t2);
+    auto t3 = t1;
+    auto t4 = t2;
+
+    t4 = onedpl_t1;
+    EXPECT_TRUE(t1 == t4, "assignment of oneDPL tuple to std::tuple provides incorrect results");
+
+    t3 = onedpl_t2;
+    EXPECT_TRUE(t2 == t3, "assignment of oneDPL tuple to std::tuple provides incorrect results");
+
+    onedpl_t3 = t1;
+    EXPECT_TRUE(onedpl_t1 == onedpl_t3, "assignment of oneDPL tuple from std::tuple provides incorrect results");
+
+    onedpl_t4 = t2;
+    EXPECT_TRUE(onedpl_t2 == onedpl_t4, "assignment of oneDPL tuple from std::tuple provides incorrect results");
+
+    decltype(onedpl_t1) onedpl_t5 = onedpl_t2;
+    decltype(onedpl_t1) onedpl_t6 = onedpl_t1;
+
+    swap(onedpl_t5, onedpl_t6);
+    EXPECT_TRUE(((onedpl_t1 == onedpl_t5) && (onedpl_t6 == onedpl_t2)),
+                "swap of oneDPL tuple provides incorrect results");
+}
+
+int
+main()
+{
+
+    test_tuple(std::tuple<int, int, int>{1, 2, 3}, std::tuple<int, int, int>{1, 2, 3});
```

Comment:
Does it make sense to add tests for Kernels too in this test?

Reply:
Do you mean calling sycl kernels or oneDPL algos with oneDPL internal tuples?  Checking `is_trivially_copyable` should check that this is possible.  Also, we have other tests which utilize `zip_iterator` with the algorithms which should provide system level testing.

My goal when adding this new test is to unit test tuple for some specific attributes we want it to have.